src/View/Form/EntityContext.php

```diff
@@ -220,7 +230,18 @@ public function val($field)
         }
 
         if ($entity instanceof EntityInterface) {
-            return $entity->get(array_pop($parts));
+            $part = array_pop($parts);
+            $val = $entity->get($part);
+            if ($val !== null) {
+                return $val;
+            }
+            if ($options['default'] !== null
+                || !$options['schemaDefault']
```

Comment:
It does. Check [this](https://github.com/cakephp/cakephp/pull/8735/files#diff-cf0be6f0051f81c7f12aaa4c6becafc7R4053) test.